client/src/components/molecules/ClientRequestTable.tsx

```diff
@@ -0,0 +1,229 @@
+import { gql, useMutation } from "@apollo/client";
+import React, { FunctionComponent, useEffect, useState } from "react";
+import Form from "react-bootstrap/Form";
+import moment from "moment";
+import Table from "react-bootstrap/Table";
+
+import Request from "../../data/types/request";
+import RequestForm from "../organisms/RequestForm";
+
+interface Props {
+  requests: Request[];
+  onChangeRequests?: (requests: Request[]) => void;
+}
+
+const ClientRequestTable: FunctionComponent<Props> = (props: Props) => {
+  const [requestSelectedForEditing, setRequestSelectedForEditing] =
+    useState("");
+
+  const headingList = [
+    "Fulfilled",
+    "Request Group",
+    "Request Type",
+    "Quantity",
+    "Date Requested",
+    "",
+  ];
+  const updateRequest = gql`
+    mutation updateRequest($request: RequestInput) {
+      updateRequest(request: $request) {
+        id
+        success
+        message
+      }
+    }
+  `;
+  const softDeleteRequest = gql`
+    mutation deleteRequest($id: ID) {
+      softDeleteRequest(id: $id) {
+        id
+        success
+        message
+      }
+    }
+  `;
+
+  const [requests, setRequests] = useState<Request[]>([]);
+  useEffect(() => {
+    const requests = props.requests.filter((request) => !request.deleted);
+    const nonFulfilledRequests = requests.filter((request) => {
+      if (request !== undefined) {
+        if (request.fulfilled === false) {
+          return request;
+        }
+      }
+    });
+    const fulfilledRequests = requests.filter((request) => {
+      if (request !== undefined) {
+        if (request.fulfilled === true) {
+          return request;
+        }
+      }
+    });
+
+    nonFulfilledRequests.sort((a, b) => {
+      return a!.dateCreated!.valueOf() - b!.dateCreated!.valueOf();
+    });
+
+    fulfilledRequests.sort((a, b) => {
+      return a!.dateCreated!.valueOf() - b!.dateCreated!.valueOf();
+    });
+
+    const sortedRequests: Request[] = nonFulfilledRequests!.concat(
+      fulfilledRequests!
+    ) as Request[];
+    setRequests(sortedRequests);
+  }, [props.requests]);
+
+  const [mutateDeleteRequest] = useMutation(softDeleteRequest);
+  const [mutateRequest] = useMutation(updateRequest);
+
+  const onSoftDeleteRequest = (index: number) => {
+    const requestsCopy = requests.slice();
+    const req = { ...requestsCopy[index] };
+    requestsCopy.splice(index, 1);
+    const id = req._id;
+    setRequests(requestsCopy);
+    props.onChangeRequests!(requestsCopy);
```

Comment:
Did you mean onChangeRequests? (question mark)

Reply:
I added the "!" to prevent Typescript complaints because `onChangeRequests` is an optional prop. I'll change it to be a required prop so that it makes more sense